### PR TITLE
Adding netty-transport-native-kqueue dependency

### PR DESCRIPTION
**Pull request by jain-shivangi**

### CODE REVIEW

File: sdk/core/azure-core-http-netty/pom.xml [hunk 1]
```diff
@@ -100,6 +100,11 @@
       <version>4.1.49.Final</version> <!-- {x-version-update;io.netty:netty-transport-native-epoll;external_dependency} -->
       <classifier>linux-x86_64</classifier>
     </dependency>
+    <dependency>
+      <groupId>io.netty</groupId>
+      <artifactId>netty-transport-native-kqueue</artifactId>
+      <version>4.1.49.Final</version> <!-- {x-version-update;io.netty:netty-transport-native-kqueue;external_dependency} -->
+    </dependency>
```
Comment (srnagar) [inline sdk/core/azure-core-http-netty/pom.xml:107]: This new dependency should also be added to the allowed list of dependencies. See line 193 for example. Also, this new external dependency has to be tracked in `eng/versioning/external_dependencies.txt`.

Comment (jain-shivangi) [reply sdk/core/azure-core-http-netty/pom.xml]: Thanks for the help. Done the changes.

File: sdk/core/azure-core-http-netty/pom.xml [hunk 2]
```diff
@@ -100,6 +100,11 @@
       <version>4.1.49.Final</version> <!-- {x-version-update;io.netty:netty-transport-native-epoll;external_dependency} -->
       <classifier>linux-x86_64</classifier>
     </dependency>
+    <dependency>
+      <groupId>io.netty</groupId>
+      <artifactId>netty-transport-native-kqueue</artifactId>
+      <version>4.1.49.Final</version> <!-- {x-version-update;io.netty:netty-transport-native-kqueue;external_dependency} -->
```
[the remark this answers is not in the document]
Comment (jain-shivangi) [reply sdk/core/azure-core-http-netty/pom.xml:106]: Thanks for the help. Done the changes here too.

**Comment by jain-shivangi** [comment]
Thanks for the approval. Kindly merge it when free.

**Pull request merged**